### DISCUSSION TITLE
Dangerous Extension of Feature Checking.

### DISCUSSION THREAD

**Discussion by polatefekaya** [abpframework/abp#21734] [Q&A] [answered]
https://github.com/abpframework/abp/blob/dc3a6d5721bd2aefb1a9ef43d0a87d1af8a6ddcd/framework/src/Volo.Abp.Features/Volo/Abp/Features/FeatureCheckerExtensions.cs#L55-L62

I found this code while reading through the documentaion. I think the naming is vague and the lackness of comment explanation makes that method dangerous for the people who doesn't read the documentation. 

Maybe a XML Comment will be useful for user to understand what that method returns. 

And I don't get why this framework needs a method that throws an exception if the value is `false`. User can simply use `IsEnabledAsync` and can throw an exception based on the return value.

**Comment by maliming** [comment, marked as answer]
hi

This method will be used in filters or interceptions. 

I will add the comment for the extension methods that will throw exceptions.

---

https://github.com/abpframework/abp/pull/21737
